src/main/java/com/example/demo/controller/FruitsController.java

```diff
@@ -0,0 +1,95 @@
+package com.example.demo.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.example.demo.model.User;
+import com.example.demo.service.FruitsService;
+
+@Controller
+@RequestMapping("/fruits")
+public class FruitsController {
+	
+	@Autowired
+	private FruitsService service;
+	
+	//select全件表示
+//    @GetMapping("/list")
+//    public String getUserList(Model model) {
+//        List<User> userList =  service.getList();
+//        model.addAttribute("fruits", userList);
+//        //model.addAttribute("message", "hello!");
+//        return "fruits/list";
+//    }
+	
+    //top.htmlを表示
+    @GetMapping("")
+    public String top(Model model, @ModelAttribute User u) {
+        model.addAttribute("fruits", service.getList());
+        model.addAttribute("message", "hello!");
+        return "fruits/top";
+    }
+
+    //select１件 詳細画面
+    @GetMapping({"/details", "details/id={id}"})
+    public String details(@PathVariable(required = false) Integer id, Model model) {
+    	
+    	if (id == null) {
+    		return "redirect:/fruits";
+		}
+        model.addAttribute("fruits", service.getUserOne(id));
+        return "fruits/details";
+    }
+
+    //新規登録画面へ遷移
+    @GetMapping("/register")
+    public String registerUser(@ModelAttribute User u, Model model) {
+        model.addAttribute("fruits", u);
+        return "fruits/register";
+    }
+    
+    //新規登録してtop画面へ遷移
+    @PostMapping("/register")
+    public String create(@ModelAttribute @Validated User u,
+    		BindingResult result, Model model) {
+        if (result.hasErrors()) {
+            return registerUser(u, model);
+        }
+        service.insertOne(u);
+        return "redirect:/fruits";
+    }
+
+
+    //変更画面へ遷移
+    @GetMapping("change/id={id}")
+    public String change(@PathVariable("id") Integer id, Model model) {
+        model.addAttribute("fruits", service.getUserOne(id));
+        return "fruits/change";
+    }
+    
+    //詳細変更をしてtop画面へ遷移（変更はhtml内でreadonlyにして、値段のみ変更可）
+    @PostMapping("change/id={id}")
+    public String update(@ModelAttribute User u, Model model) {
+        service.updateOne(u.getId(), u.getName(), u.getPrice());
+        return "redirect:/fruits";
+    }
+
+    //削除してtop画面に遷移
+    @PostMapping({"/delete", "delete/id={id}"})
+    public String delete(@PathVariable(required = false) Integer id) {
+    	
+    	if (id == null) {
+    		return "redirect:/fruits";
+		}
```

Comment:
こちらのソースですが
```suggestion
    @PostMapping("/delete")
    public String delete(@PathVariable Integer id) {
```
のようにするのがいいかな？と思いました。

理由としては正常に画面を操作していてidがnullで来ることはないからです。
仮にidがnullで来るならばシステムエラーとするのが適切ですね。

またエラー画面ですが今だとデフォルトのエラー画面になると思いますので以下を参考にしてエラー画面をカスタマイズしてみるのも勉強になると思います。
エラー処理というのは普通に勉強していると見逃しがちなのですがすごく重要です！！！ 🚀 
参考
https://www.baeldung.com/spring-boot-custom-error-page

調べる際のキーワードとしては
- spring boot
- thymeleaf
- error.html
- errorcontroller
などですね！

Reply:
両方修正してみたいと思います。

Reply:
はい！よろしくおねがいします！

Reply:
レビュー後の修正をしました。
c4266d3
- delete、detaisメソッドのmapping部分の変更
- 上記メソッドのテスト部分を少し修正
- エラーページを作成（共通エラー、404エラー）、ErrorControllerの作成


mappingのところでは、idを紐づくようにしたいので
@PostMapping("/delete/{id}") もしくは ("/{id}")となるよう考え、わかりやすいように前者のほうにしました。

再度レビューお願いします。

Reply:
@hiroyukisss1 
再レビューしてコメントしました！